python/hail/linalg/rowmatrix.py

```diff
@@ -0,0 +1,251 @@
+from hail.utils.java import Env, joption
+from hail.typecheck import *
+from hail.expr.expressions import expr_float64, matrix_table_source, check_entry_indexed
+
+row_matrix_type = lazy()
+
+
+class RowMatrix(object):
+    """Hail's row-distributed matrix of :py:data:`.tfloat64` elements.
+
+    .. include:: ../_templates/experimental.rst
+
+    Notes
+    -----
+    Row matrices must have fewer than :math:`2^31` columns.
+
+    Under the hood, row matrices are partitioned into group of rows.
```

Comment:
should be `groups of rows.`

Reply:
Also, is is this information necessary to the user?

Reply:
I think its a reasonable to let the user know how its different from BlockMatrix and because partition_size is a parameter in some methods (just as block_size appears in BlockMatrix). lmm2 requires two row matrices with the same partitioning so they can be zipped together.